### DISCUSSION TITLE
version information

### DISCUSSION THREAD

**Discussion by sbeckeriv** [go-shiori/shiori#766] [Q&A] [answered]
Dearest Maintainer, 

I am running Shiori in a docker. I dont seem to have path /api/v1/auth/login which got me wondering what version of shiori am I running. A command line -V or even in the settings ui would be nice. 

Thanks for reading and thanks for the great program.

**Comment by Monirzadeh** [comment, marked as answer]
Hi
` /api/v1/auth/login` is related to new API we work on that https://github.com/go-shiori/shiori/issues/640. it is not actual shiori version.
@fmartingr  added shiori version in command line and login page in [v1.6.0-rc.1](https://github.com/go-shiori/shiori/releases/tag/v1.6.0-rc.1) (not stable release yet)
if you don't see them so you are on  on last stable release [v1.5.5](https://github.com/go-shiori/shiori/releases/tag/v1.5.5) or lower.
for more information you can see https://github.com/go-shiori/shiori/issues/718#issuecomment-1712735196

**Comment by pocelka** [comment]
@Monirzadeh / @fmartingr do you think that it would be possible to have the version indicator enhanced little bit to show also if there is a newer version available? I.e. current (new) while the version text could contain also link to repository so user can quickly check release notes?

**Reply by fmartingr** [reply]
That's a nice idea, we would need an easy way for a shiori instance to check the latest available version, unsure if we can do that directly with github without authentication. We could research that at some point.

**Reply by pocelka** [reply]
Should be possible and quite easy if you can process json in go…

https://api.github.com/repos/go-shiori/shiori/releases

also I’m not really sure that having version on login page is good idea. Because if you are logged in with “remember me” there is some timeout until you get disconnected. It would be better in my opinion to have version indicator on home page, I.e footer section.

**Reply by fmartingr** [reply]
> https://api.github.com/repos/go-shiori/shiori/releases

Awesome, didn't test that was accessible publicly, then it should be relatively easy. We need a way for shiori to run arbitrary background jobs and a way to check if we want to check for release candidates too. It's not super hard, but it piles up with all the other things.

> also I’m not really sure that having version on login page is good idea. Because if you are logged in with “remember me” there is some timeout until you get disconnected. It would be better in my opinion to have version indicator on home page, I.e footer section.

Yeah, I'm not a fan of having it there because security reasons (though it get exposed via an api call anyway), but didn't find a good place to put it in the main UI back in the day. Once we have that in place we can iterate forward. With an update check, we could add that to the settings page for admins, for example, showing your current version + if there's any updates.
